File: LeetCode/is_subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(!s.size() && !t.size())
            return true;
        
        int j = 0;
        
        for(int i = 0; i < t.size(); i++) {
            if(s[j] == t[i])
                j++;
            if(j == s.size())
                return true;
        }
        
        return false;
    }
};

File: LeetCode/is_subsequence.go
func isSubsequence(s string, t string) bool {
    if len(s) == 0 {
        return true
    }

    if len(s) > len(t) {
        return false
    }

    j := 0
    for i := 0; i < len(t); i++ {
        if s[j] == t[i] {
            j++
        }

        if j == len(s) {
            return true
        }
    }

    return false
}
